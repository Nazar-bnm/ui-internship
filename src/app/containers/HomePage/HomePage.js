import React, {Component} from 'react';

class HomePage extends Component {
  render() {
    return <h2>HomePage</h2>
  }
}

export default HomePage