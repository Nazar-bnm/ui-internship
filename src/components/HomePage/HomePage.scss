@import '../../scss/global_styles';

.mainPage {
  @include flex($justify: space-between);
  width: 100%;
  padding-top: 50px;

  .heading__wrapper {
    margin: 20px 0 50px;
    font-size: 1.2rem;
  }

  &__popular {
    width: 68%;
  }

  &__blog {
    height: 50%;
    width: 30%;
    border: 1px solid $mercury;
    padding: 20px 23px;
  }
}

@include media-for(tablet) {
  .mainPage {
    &__popular {
      width: 100%;
    }

    &__blog {
      width: 100%;
      padding: 30px;
    }
  }
}
