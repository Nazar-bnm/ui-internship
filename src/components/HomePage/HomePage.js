import React from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import WhatIsNew from '../WhatIsNew';
import BlogArticlePreview from '../BlogArticlePreview';
import MostPopular from '../MostPopular';
import Slideshow from './Slideshow';
import Slide from './Slide';
import Heading from '../Heading';
import { ANIMATION_NAMES } from '../../constants/SlideshowConst';
import { slideshowData } from './Slideshow/SlideshowDate';

import './HomePage.scss';

const CN = 'homepage';

const HomePage = ({ className }) => {
  const slidesData = slideshowData.map((el) => ({
    id: el.id,
    img: el.img,
    component: (
      <Slide
        title={el.title}
        description={el.description}
        buttonName={el.buttonName}
      />
    )
  }));

  return (
    <div className={cx(CN, className)}>
      <Slideshow animation={ANIMATION_NAMES.ZOOM_IN} slideData={slidesData} />
      <div>
        <WhatIsNew />
        <div className="container content mainPage">
          <div className="mainPage__popular">
            <Heading title="most popular" position="center" />
            <MostPopular />
          </div>
          <div className="mainPage__blog">
            <BlogArticlePreview />
          </div>
        </div>
      </div>
    </div>
  );
};

HomePage.propTypes = {
  className: PropTypes.string
};

HomePage.defaultProps = {
  className: ''
};

export default HomePage;
