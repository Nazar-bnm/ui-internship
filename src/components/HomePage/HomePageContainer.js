import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { getProductsError, getProducts, getProductsPending } from '../../reducers/reducerAPIcall';
import fetchProductsAction from '../../actions/fetchProducts';

import HomePage from './HomePage';

const mapStateToProps = (state) => ({
  error: getProductsError(state),
  products: getProducts(state),
  pending: getProductsPending(state)
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  fetchProducts: fetchProductsAction
}, dispatch);

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomePage);
