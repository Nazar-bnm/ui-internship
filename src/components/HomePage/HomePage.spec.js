import React from 'react';
import { shallow } from 'enzyme';
import HomePage from './HomePage';

describe('<HomePage />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <HomePage />
    );
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
