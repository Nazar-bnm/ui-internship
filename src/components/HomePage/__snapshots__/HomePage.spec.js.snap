// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should match snapshot 1`] = `
<div>
  <Slideshow
    animation="zoom-in"
    className=""
    slideData={
      Array [
        Object {
          "component": <Slide
            buttonName="discover the editorial"
            className=""
            description="Special autumn edition"
            title="new collection"
          />,
          "id": 1,
          "img": "",
        },
        Object {
          "component": <Slide
            buttonName="find a perfect outfit"
            className=""
            description="a long journey"
            title="new collection"
          />,
          "id": 2,
          "img": "",
        },
        Object {
          "component": <Slide
            buttonName="start shopping now"
            className=""
            description="a new look"
            title="sales"
          />,
          "id": 3,
          "img": "",
        },
        Object {
          "component": <Slide
            buttonName="find a perfect outfit"
            className=""
            description="a new wave"
            title="new collection"
          />,
          "id": 4,
          "img": "",
        },
      ]
    }
  />
  <ProductList />
</div>
`;
